### PR TITLE
- Drop support for Python 2.7, 3.5, 3.6

### PR DESCRIPTION
**Pull request by dataflake**

### CODE REVIEW

File: src/zope/i18nmessageid/message.py [hunk 1]
```diff
@@ -66,7 +64,7 @@ def __new__(cls, ustr, domain=_marker, default=_marker, mapping=_marker,
             self.number = number
 
         if self.number is not None and not isinstance(
-                self.number, six.integer_types + (float,)):
+                self.number, (int,) + (float,)):
```
Comment (icemac) [inline src/zope/i18nmessageid/message.py]:
```suggestion
                self.number, (int, float)):
```

File: tox.ini [hunk 2]
```diff
@@ -4,15 +4,11 @@
 minversion = 3.18
```
Comment (icemac) [inline tox.ini:4]: I think we should update this to `minversion = 4.0` in `meta/config`, so developers do not run into problem when using `tox < 4` locally.

**Review by icemac — APPROVED** [review]
LGTM besides some minor suggestions.

File: tox.ini (hunk 2 above)
Comment (dataflake) [reply tox.ini:4]: Created PR https://github.com/zopefoundation/meta/pull/188

**Pull request merged**